mistralrs-core/src/sequence.rs

```diff
@@ -243,21 +249,45 @@ impl Sequence {
         }
         self.state.set(state);
     }
+    fn is_sub<T: PartialEq>(mut haystack: &[T], needle: &[T]) -> bool {
+        if needle.is_empty() {
+            return true;
+        }
+        while !haystack.is_empty() {
+            if haystack.starts_with(needle) {
+                return true;
+            }
+            haystack = &haystack[1..];
+        }
```

Comment:
Maybe it's worth it to use a more optimized version of this is_sub?

eg: https://docs.rs/galil-seiferas/latest/galil_seiferas/ from https://stackoverflow.com/a/47045355

Reply:
I added it...

Reply:
I'm not sure if this is the best library...